api/utils.ts

```diff
@@ -0,0 +1,21 @@
+export function allDoNotContainDangerousCharacters(values: string[]) {
+  for (const value of values) {
+    if (!hasNoDangerousCharacters(value)) {
+      // if one value contains dangerous characters, return false
+      return false;
+    }
+  }
+  return true;
+}
+
+export function hasNoDangerousCharacters(value: string): boolean {
```

Comment:
I don't like this function name. Ideally it would be called hasUnacceptableChars and you would negate the return value. 
It is harder for humans to reason about a function called "notDoX" than a function called "DoX"
You also need to add in a comment why those chars are not accepted.

Reply:
Ok can do that